pkg/controllers/management/clustertemplate/revision_controller.go

```diff
@@ -0,0 +1,85 @@
+package clustertemplate
+
+import (
+	"context"
+
+	"fmt"
+	"strings"
+
+	"github.com/rancher/rancher/pkg/namespace"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	RevisionController   = "mgmt-cluster-template-revision-controller"
+	clusterTemplateLabel = "io.cattle.field/clusterTemplateId"
+)
+
+type RevController struct {
+	clusterTemplates              v3.ClusterTemplateInterface
+	clusterTemplateLister         v3.ClusterTemplateLister
+	clusterTemplateRevisions      v3.ClusterTemplateRevisionInterface
+	clusterTemplateRevisionLister v3.ClusterTemplateRevisionLister
+}
+
+func newRevController(ctx context.Context, mgmt *config.ManagementContext) *RevController {
+	n := &RevController{
+		clusterTemplates:              mgmt.Management.ClusterTemplates(namespace.GlobalNamespace),
+		clusterTemplateLister:         mgmt.Management.ClusterTemplates(namespace.GlobalNamespace).Controller().Lister(),
+		clusterTemplateRevisions:      mgmt.Management.ClusterTemplateRevisions(namespace.GlobalNamespace),
+		clusterTemplateRevisionLister: mgmt.Management.ClusterTemplateRevisions(namespace.GlobalNamespace).Controller().Lister(),
+	}
+	return n
+}
+
+func Register(ctx context.Context, management *config.ManagementContext) {
+	n := newRevController(ctx, management)
+	if n != nil {
+		management.Management.ClusterTemplateRevisions("").AddHandler(ctx, RevisionController, n.sync)
+	}
+}
+
+//sync is called periodically and on real updates
+func (n *RevController) sync(key string, obj *v3.ClusterTemplateRevision) (runtime.Object, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	if obj.Spec.ClusterTemplateName != "" {
+		//load the template
+		split := strings.SplitN(obj.Spec.ClusterTemplateName, ":", 2)
+		if len(split) != 2 {
+			return nil, fmt.Errorf("error in splitting clusterTemplate name %v", obj.Spec.ClusterTemplateName)
+		}
+		templateName := split[1]
+		template, err := n.clusterTemplates.Get(templateName, metav1.GetOptions{})
+		if err != nil {
```

Comment:
Do we want to return error even if the clustertemplate is not found?

Reply:
yes, since we are trying to update the template's defaultRevision reference